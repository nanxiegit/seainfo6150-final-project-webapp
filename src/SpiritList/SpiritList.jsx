import React from "react";
import PropTypes from "prop-types";
import SpiritListItem from "../SpiritListItem/SpiritListItem.jsx";

import styles from "./SpiritList.module.css";

const SpiritList = (props) => {
  let displayContent;

  if (props.Recipes.length) {
    displayContent = (
      <ul className={styles.container}>
        {props.Recipes.map((recipe) => (
          <SpiritListItem recipe={recipe} key={recipe.slug} />
        ))}
      </ul>
    );
  } else {
    displayContent = <div>You have no data!</div>;
  }

  return (
    <div>
      <h2 className={styles.header}>
      {props.Recipes[0].type} 
      </h2>
      {displayContent}
    </div>
  );
};

SpiritList.propTypes = {
    Recipes: PropTypes.array.isRequired
}
export default SpiritList;