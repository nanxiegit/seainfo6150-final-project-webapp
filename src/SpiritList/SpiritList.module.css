.container {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    gap: 20px;
    font-family: Fantasy;
    list-style-type: none;
    padding: 40px;
    background-color: rgb(180, 183, 186);
  }

  .header{
    font-family: Copperplate;
    background-color: rgb(180, 183, 186);
    font-size: 45px;
    text-align: center;
    color:rgb(88, 88, 88);
    text-decoration: underline overline dotted;
  }

  @media only screen and (max-width: 900px) {
    .container  {
      grid-template-columns: repeat(1, 1fr);
    }
  }
  
  @media only screen and (min-width: 900px)  {
    .container  {
      grid-template-columns: repeat(2, 1fr);
    }
  }

