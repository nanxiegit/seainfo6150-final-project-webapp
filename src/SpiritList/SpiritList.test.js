import React from "react";
import SpiritList from "./SpiritList";
const Recipes = [
  {
    "slug": "Manhattan",
    "type":"bourbon",
    "title": "Manhattan",
    "text":"Ingredients<ul style='list-style-type:square' ><li>2 ounces bourbon or rye </li><li>1 ounce sweet vermouth</li><li> 2 dashes Angostura bitters</li><li> 1 dash orange bitters </li><li>Garnish: brandied cherry</li></ul> Steps<ul style='list-style-type:decimal'><li>Add all the bourbon (or rye), sweet vermouth and both bitters to a mixing glass with ice, and stir until well-chilled. </li><li>Strain into a chilled coupe. </li><li>Garnish with a brandied cherry.</li></ul>",
    "shortText":"Date Night",
    "timeStamp": "2020-06-29T15:12:24.000Z",
    "pubDate": "Thu, 29 Jun 2020 15:12:24 GMT",
    "image": {
            "_url": "https://www.liquor.com/thmb/gSN7fXX_rGkGi-3M6bmPhE7BqAk=/960x0/filters:no_upscale():max_bytes(150000):strip_icc():format(webp)/__opt__aboutcom__coeus__resources__content_migration__liquor__2018__05__10144903__Manhattan-720x720-recipe-9497922907c14d91898f557cb51f2ea3.jpg",
            "_height": "619",
            "_width": "1100"
}
  }
];

describe("SpiritList tests", () => {
  it("renders correctly", () => {
    const { container } = render(<SpiritList Recipes ={Recipes} />);
    expect(container).toMatchSnapshot();
  });
});