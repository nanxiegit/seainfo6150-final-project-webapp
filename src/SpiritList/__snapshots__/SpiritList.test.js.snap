// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpiritList tests renders correctly 1`] = `
<div>
  <div>
    <h2
      class="header"
    >
      bourbon
    </h2>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <article
          class="article"
        >
          <header
            class="header"
          >
            <img
              alt="Manhattan"
              class="image"
              src="https://www.liquor.com/thmb/gSN7fXX_rGkGi-3M6bmPhE7BqAk=/960x0/filters:no_upscale():max_bytes(150000):strip_icc():format(webp)/__opt__aboutcom__coeus__resources__content_migration__liquor__2018__05__10144903__Manhattan-720x720-recipe-9497922907c14d91898f557cb51f2ea3.jpg"
            />
            <h2
              class="title"
            >
              <a
                class="link"
                href="/bourbon/Manhattan"
              >
                Manhattan
              </a>
            </h2>
          </header>
          <div
            class="wrapper"
          />
          <button
            class="button"
          >
            Sip Hint
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
